### PR TITLE
Infer the proxy user automatically for delete batch operation

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description
Infer the batch user from session or metadata, user do not need to specify the proxy user anymore. 

This pr also align the behavior of BatchesResource with that of SessionsResource and OperationsResource(no proxy user parameter).

For Kyuubi Batch, Session and Operation, these resources have the user attiribute.

So we only need to check whether the authentication user has the permission to access the resource.

## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [ ] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793 @cxzl25

**Comment by pan3793** [comment]
I think it may be useful in the future.

Currently, we only have a true-false-switch for roles: admin/non-admin. But in the future, we may introduce the real RBAC system, which means we can control `tom` only allows proxying `jerry`, as Hadoop does.

File: docs/deployment/migration-guide.md [hunk 1]
```diff
@@ -32,6 +32,8 @@
   relative to `$KYUUBI_HOME` instead of `$PWD`.
 * Since Kyuubi 1.8, PROMETHEUS is changed as the default metrics reporter. To restore previous behavior,
   set `kyuubi.metrics.reporters=JSON`.
+* Since Kyuubi 1.8.1, for `DELETE /batches/${batchId}`, `proxyUser` and `hive.server2.proxy.user`
```
Comment (pan3793) [inline docs/deployment/migration-guide.md:35]: this should be listed in

```
## Upgrading from Kyuubi 1.8.0 to 1.8.1
```

but seems that change is not necessary to be mentioned

File: kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/BatchCliArgumentsSuite.scala [hunk 2]
```diff
@@ -119,7 +119,7 @@ class BatchCliArgumentsSuite extends KyuubiFunSuite with TestPrematureExit {
     }
   }
 
-  test("delete batch with hs2ProxyUser") {
+  ignore("delete batch with hs2ProxyUser") {
```
Comment (pan3793) [inline kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/BatchCliArgumentsSuite.scala:122]: delete directly?

File: kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/BatchRestApi.java [hunk 3]
```diff
@@ -101,6 +101,11 @@ public OperationLog getBatchLocalLog(String batchId, int from, int size) {
     return this.getClient().get(path, params, OperationLog.class, client.getAuthHeader());
   }
 
+  /**
+   * hs2ProxyUser for delete batch is deprecated since 1.8.1, please use {@link
+   * #deleteBatch(String)} instead.
+   */
+  @Deprecated
   public CloseBatchResponse deleteBatch(String batchId, String hs2ProxyUser) {
     Map<String, Object> params = new HashMap<>();
```
Comment (pan3793) [inline kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/BatchRestApi.java:110]: log a deprecated warning message here

**Comment by turboFei** [comment]
have addressed all comments and updated the pr description. @pan3793

File: docs/deployment/migration-guide.md [hunk 4]
```diff
@@ -21,6 +21,11 @@
 
 * Since Kyuubi 1.9.0, `kyuubi.session.conf.advisor` can be set as a sequence, Kyuubi supported chaining SessionConfAdvisors.
 
+## Upgrading from Kyuubi 1.8.0 to 1.8.1
+
+* Since Kyuubi 1.8.1, for `DELETE /batches/${batchId}`, `proxyUser` and `hive.server2.proxy.user`
```
Comment (pan3793) [inline docs/deployment/migration-guide.md:26]: proxyUser only present in master

**Review by pan3793 — APPROVED** [review]
LGTM, directly cherry-pick to branch-1.8 may have conflicts

**Pull request closed without merging**